docs/adrs/0020-pipeline-academies-from-adb.md

```diff
@@ -0,0 +1,115 @@
+
+To meet our deadlines, we propose an interim approach: leveraging the existing synchronisation mechanisms where these applications sync their data back to the Academies Database daily for reporting purposes. This allows us to utilise a consolidated and up-to-date data source for our needs while maintaining acceptable performance and reliability.  
+
+## Decision  
+
+For the interim, we will integrate data from the Academies Database (ADB) to support the "Pipeline Academies" feature. This approach involves querying specific tables within ADB, which aggregate data synced from the relevant applications.  
+
+In the medium term, we will transition to consuming this data via versioned APIs from the respective applications once they become available.  
+
+## Reasons for the Decision  
+
+1. **Access to Current Data**: Leveraging the ADB ensures we have access to data that is synced daily from the relevant applications, enabling us to work with up-to-date information.  
+2. **Simplicity and Feasibility**: This approach avoids dependencies on external teams delivering APIs within our tight timelines, reducing implementation risk.  
+3. **Performance**: Consolidating data into a single "super atomic" call from the ADB ensures our application remains performant, while allowing us to query the exact data we need.  
+4. **Maintainability**: While interim integration requires managing changes to ADB schema, transitioning to APIs later will improve maintainability with version control and documentation inherent to API usage.  
+
+## Consequences  
+
+### Short-Term Benefits  
+
+- **Rapid Development**: We can meet project deadlines without waiting for API availability.  
+- **Centralised Data Access**: ADB provides a unified source of truth for data from multiple applications.  
+
+### Drawbacks  
+
+1. **Dependence on ADB Schema**: Changes in the data model or structure of contributing applications may impact our queries and require maintenance.  
+2. **Limited Data Validation**: Relying on ADB means less control over the accuracy of the data, as it is aggregated and not fetched directly from source systems.  
+
+### Medium-Term Considerations  
+
+- Transitioning to APIs will provide improved control, validation, and resilience against schema changes.  
+
+## Trade-offs  
+
+### Simplified Implementation vs. Future Maintenance  
+
+This decision simplifies our immediate implementation but introduces potential maintenance overhead as ADB schemas evolve with changes in upstream applications.  
+
+### Interim vs. Long-Term Solutions  
+
+Our interim reliance on ADB data ensures we can deliver the feature on time but will require effort to migrate to API-based integration later.  
+
+## Requirements for API Integration (Medium-Term)  
+
+While our analysis is ongoing, the following data fields are identified as necessary for the integration:  
```

Comment:
Awesome seeing the fields we intend to use in the ADR.
Do these new academies db tables have primary keys and/or indexed columns we can use to identify a record in the new tables?

Reply:
They are exact mirrors of what the products themselves are utilising in their own databases, there will be ongoing analysis and conversations with the relevant teams to make sure we have identifiers we need. Having looked at it all of them seem to have a trust link (albeit this changes dependant on product) so either ukprn, trn etc and where a trust id of some kind isn't available it's because the data is not appropriate, i.e., the early stages of Form a MAT for example in prepare where a trust isn't connection to the conversion as of yet.